database/src/migrations/20220602152015_add_record_sims_eml_transformation.ts

```diff
@@ -15,40 +15,138 @@ export async function up(knex: Knex): Promise<void> {
-    , fundings as (select f.proj_n, f.project_type, f.fund_n, fs.funds_n, jsonb_build_object('agencyName', fs.fundings->'para', 'fundingStartDate', jsonb_path_query_first(fs.fundings, '$.section[*] \\? (@.title == "Funding Start Date").para'), 'fundingEndDate', jsonb_path_query_first(fs.fundings, '$.section[*] \\? (@.title == "Funding End Date").para')) funding_object from funding f, jsonb_array_elements(funding) with ordinality fs(fundings, funds_n))
-    , funding_arrs as (select proj_n, project_type, jsonb_agg(funding_object) funding_arr from fundings group by project_type, proj_n)
-    , project_objects as (select jsonb_build_object('projectId', aps.project->'@_id'
-      , 'projectType', aps.project_type
-      , 'projectTitle', aps.project->'title'
-      , 'projectOrganizationName', aps.project->'personnel'->'organizationName'
-      , 'projectAbstract', jsonb_path_query_first(aps.project, '$.abstract.section')
-      , 'taxonomicCoverage', jsonb_path_query_first(aps.project, '$.studyAreaDescription.coverage.taxonomicCoverage')
-      , 'fundingSource', fas.funding_arr
-		) project_object from all_projects aps
-      	left join funding_arrs fas on fas.proj_n = aps.proj_n and fas.project_type = aps.project_type
-     	)
-    , project_arr as (select jsonb_agg(project_object) project_array from project_objects)
-    select jsonb_strip_nulls(jsonb_build_object('datasetTitle', d.dataset->'title'
-      , 'datasetId', d.dataset->'@_id'
-      , 'sourceSystem', d.dataset->'@_system'
-      , 'publishDate', d.dataset->'pubDate'
-      , 'project', p.project_array
-      , 'projectIUCNConservationActions', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.IUCNConservationActions.IUCNConservationAction')
-      , 'projectStakeholderPartnerships', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.stakeholderPartnerships.stakeholderPartnership')
-      , 'projectActivities', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.projectActivities.projectActivity')
-      , 'projectClimateChangeInitiatives', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.projectClimateChangeInitiatives.projectClimateChangeInitiative')
-      , 'projectFirstNations', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.firstNationPartnerships.firstNationPartnership')
-      , 'projectSurveyProprietors', jsonb_path_query_first(e.eml, '$.additionalMetadata[*].metadata.projectSurveyProprietors.projectSurveyProprietor')
-      )) result_data from eml e, datasets d, project_arr p;$transform$);
-
+    insert into source_transform (
+      system_user_id, 
+      version, 
+      metadata_index, 
+      metadata_transform
+    ) values (
+      (select system_user_id from system_user where user_identifier = 'service-account-SIMS-SVC-4464'), 
+      '1.0', 
```

Comment:
is this a version number for the transformation? should it be incremented so we know which transform is which?

Reply:
I don't think we're getting past version 1.0 yet